### ISSUE TITLE
Variables names from within role should not use role name as prefix when using ansible connection variables

### ISSUE THREAD

**Issue by Johann-Zurner** [ansible/ansible-lint#5099]
##### Summary

Since ansible-lint 26.6.0 or possibly earlier, variable names from within roles must use the role name as a prefix.
This should not be the case when using ansble connection variables that can't be renamed.
E.g. `ansible_ssh_common_args`.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```
ansible-lint 26.6.0 using ansible-core:2.21.0 ansible-compat:26.3.0
```

##### STEPS TO REPRODUCE

```
- name: 'Switch to strict host key checking'
  ansible.builtin.set_fact:
    ansible_ssh_common_args: >-
      -o StrictHostKeyChecking=yes
  changed_when: true
```

##### Desired Behavior

Linter shouldn't report error.

##### Actual Behavior

Linter reports error
```
Variables names from within roles should use role_name_ as a prefix. (set_fact: ansible_ssh_common_args)ansible-lintvar-naming[no-role-prefix]
```
